pkg/chunk/aws/sse_config.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"flag"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	// SSEKMS config type constant to configure S3 server side encryption using KMS
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html
+	SSEKMS     = "SSE-KMS"
+	sseKMSType = "aws:kms"
+	// SSES3 config type constant to configure S3 server side encryption with AES-256
+	// https://docs.aws.amazon.com/AmazonS3/latest/dev/UsingServerSideEncryption.html
+	SSES3     = "SSE-S3"
+	sseS3Type = "AES256"
+)
+
+// SSEParsedConfig configures server side encryption (SSE)
+// struct used internally to configure AWS S3
+type SSEParsedConfig struct {
+	ServerSideEncryption string
+	KMSKeyID             *string
+	KMSEncryptionContext *string
+}
+
+// SSEConfig configures S3 server side encryption
+// struct that is going to receive user input (through config file or CLI)
+type SSEConfig struct {
+	Type                 string     `yaml:"type"`
+	KMSKeyID             string     `yaml:"kms_key_id"`
+	KMSEncryptionContext chunk.Tags `yaml:"kms_encryption_context"`
```

Comment:
I don't think we should use `chunk.Tags` here. I would simplify it just taking in input the context json as a `string`. Moreover, I would clarify in the CLI flag description that it expects a JSON as a string.